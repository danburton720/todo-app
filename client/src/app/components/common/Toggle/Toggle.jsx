import React from 'react';

export default class Toggle extends React.Component {
    render() {
        return (
            <div
                onClick={this.props.onToggle}
            >
                <div>
                    <div/>
                </div>
            </div>
        )
    }
}