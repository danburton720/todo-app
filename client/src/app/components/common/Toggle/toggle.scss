.toggleWrapper {
  cursor: pointer;
  padding: 8px;
}

.toggle {
  cursor: pointer;
  width: 18px;
  height: 18px;
  border-radius: 9px;
  border: 1px solid #c4c4c4;
  box-sizing: border-box;
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  transition: 0.1s ease-in-out;
}

.innerToggle {
  height: 10px;
  width: 10px;
  border-radius: 5px;
  background: transparent;
  transition: 0.1s ease-in-out;
}

.toggleWrapper:hover, .toggleWrapper.active {
  .toggle {
    border-color: #0066FF;
  }

  .innerToggle {
    background: #0066FF;
  }
}

.toggleWrapper.active:hover {
  .toggle {
    border-color: #0099ff;
  }

  .innerToggle {
    background: #0099ff;
  }
}