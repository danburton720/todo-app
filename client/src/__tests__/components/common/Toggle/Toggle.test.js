import React from 'react';
import {shallow} from 'enzyme';
import Toggle from '../../../../app/components/common/Toggle/Toggle';

describe('Toggle', () => {
    let shallowWrapper;
    let props;
    const wrapper = () => {
        if (!shallowWrapper) {
            shallowWrapper = shallow(<Toggle {...props}/>);
        }
        return shallowWrapper;
    };

    beforeEach(() => {
        props = {
            enabled: false,
            onToggle: jest.fn()
        };
        shallowWrapper = undefined;
    });

    describe('render', () => {
        it('should return a toggle wrapper div with some child divs', () => {
            expect(wrapper()).toMatchSnapshot();
        });
        it('should return a toggle wrapper div which has a class name of active along with some child divs', () => {
            props.enabled = true;
            expect(wrapper()).toMatchSnapshot();
        });
    });

    describe('functionality', () => {
        describe('div (.toggleWrapper)', () => {
            describe('onClick', () => {
                it('should call the prop onToggle', () => {
                    expect(props.onToggle).not.toHaveBeenCalled();
                    wrapper().find('div').first().props().onClick();
                    expect(props.onToggle).toHaveBeenCalledTimes(1);
                });
            });
        });
    });
});