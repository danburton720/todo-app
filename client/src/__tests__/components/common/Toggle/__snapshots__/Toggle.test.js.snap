// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle render should return a toggle wrapper div which has a class name of active along with some child divs 1`] = `
<div
  className="toggleWrapper active"
  onClick={[MockFunction]}
>
  <div
    className="toggle"
  >
    <div
      className="innerToggle"
    />
  </div>
</div>
`;

exports[`Toggle render should return a toggle wrapper div with some child divs 1`] = `
<div
  className="toggleWrapper"
  onClick={[MockFunction]}
>
  <div
    className="toggle"
  >
    <div
      className="innerToggle"
    />
  </div>
</div>
`;
